src/Npgsql/Internal/TypeHandlers/NumericHandlers/Int32Handler.cs

```diff
@@ -91,6 +98,8 @@ public int ValidateAndGetLength(double value, NpgsqlParameter? parameter)
     public void Write(double value, NpgsqlWriteBuffer buf, NpgsqlParameter? parameter)       => buf.WriteInt32((int)value);
     /// <inheritdoc />
     public void Write(decimal value, NpgsqlWriteBuffer buf, NpgsqlParameter? parameter)      => buf.WriteInt32((int)value);
+    /// <inheritdoc />
+    public void Write(Enum value, NpgsqlWriteBuffer buf, NpgsqlParameter? parameter)         => buf.WriteInt32(Convert.ToInt32(value));
```

Comment:
What happens if I provide an enum that's actually a long? Should we maybe have a check in validation and throw for that? We may want to even throw for a short enum, i.e. only allow writing the enum as its exact underlying integral type, to be on the safe side.

Reply:
Yeah, makes sense to add validation just in case since we can't guarantee that we'll be able to write `long` enum as `int`.

Reply:
Though on the other hand we do allow that for `float`, `long` and `double`...